app/views/users/_form.html.erb

```diff
@@ -33,7 +33,7 @@
   <%= hidden_field_tag(:from_cart, false) %>
   <%= hidden_field_tag(:partial_to_render, 'form') %>
 
-  <%= render partial: 'terms_of_service/user_accepted', locals: {f: f} unless action_name == 'new' %>
+  <%= render partial: 'terms_of_service/user_accepted', locals: {f: f} if (current_user == @user || !current_user) %>
```

Comment:
By imposing this logic on the `render` call, we don't ever get to see the ToS checkbox if it's disabled, e.g. as an admin viewing someone else's edit profile page. Shouldn't we just always render it?